src/components/MerchantPayout/VoucherStatusModal/VoucherStatusModal.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { FunctionComponent } from "react";
+import {
+  View,
+  StyleSheet,
+  Modal,
+  TouchableWithoutFeedback
+} from "react-native";
+import { LoadingCard } from "./LoadingCard";
+import { InvalidCard } from "./InvalidCard";
+import { color, size } from "../../../common/styles";
+
+const styles = StyleSheet.create({
+  background: {
+    ...StyleSheet.absoluteFillObject,
+    backgroundColor: color("grey", 100),
+    opacity: 0.8
+  },
+  cardWrapper: {
+    padding: size(3),
+    flex: 1,
+    alignItems: "center",
+    justifyContent: "center"
+  }
+});
+
+type VoucherStatuses = "CHECKING" | "INVALID" | "VALID";
```

Comment:
One way we can improve testability and encapsulation is to move the checking of voucher status into our own custom hook. With the hook, any consumer can call `checkValidity` can be updated on the `checkValidityStatus` and the eventual error or success result. The consumer can listen in on any of the changes and react accordingly. This will allow us to test that specific hook on its own without relying on the UI portions. Let me know if you're unsure.

Reply:
Okay! So should the `VoucherStatus {
  status: VoucherStatuses;
  errorMessage?: string;
  errorTitle?: string;
}` be in the hook as well or just `VoucherState = "CHECKING" | "INVALID" | "VALID";`? Also, will the validation of the voucher (from util) be called in the hook or on the UI side?

Reply:
The validation of the voucher from util can be called in the hook too!

As a consumer of this hook, I want to know the following:
```
type CheckValidityHook {
  checkValidityState: "DEFAULT" | "VALIDATING_ID" | "CHECKING_VALIDITY" | "RESULT_RETURNED";
  checkValidity: () => void;
  result: unknown; // need to figure out the type of this
  error?: Error;
  reset: () => void;
}
```

Then the consumer can call `checkValidity()` and react to any of these changes as they please.

This is just one way of looking at it!! I've used it and it works well, but feel free to explore!